src/dialects/abstract/query-generator.d.ts

```diff
@@ -144,6 +149,10 @@ export class AbstractQueryGenerator {
     options?: ArithmeticQueryOptions,
   ): string;
 
+  showIndexesQuery(tableName: TableName): string;
+
+  dropTableQuery(tableName: TableName, options?: DropTableQueryOptions): string;
```

Comment:
Do we want to keep this here as is or update to `dropTableQuery(tableName: TableName): string;`? Since the only option is `cascade` and that's only supported in postgres so we can also move that to the `PostgresQueryGenerator`.

Reply:
I think it's better if all options are declared in abstract, that's the only type users are going to get when accessing `sequelize.queryGenerator`. Then each dialect can use `rejectInvalidOptions` to let the user know whether an option is actually supporter or not

Reply:
Yeah, I wasn't sure if users can use typings of dialect specific query generators. But if that's more difficult I would agree that `rejectInvalidOptions` is the way to go